File: src/main/java/com/darrenfinch/feastyweb/meal/models/Meal.java
package com.darrenfinch.feastyweb.meal.models;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Data
@Entity
@Table(name = "meal")
public class Meal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name="user_id")
    private Long userId;

    @Column(name = "title")
    private String title;

    @JsonManagedReference
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "meal", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MealFood> mealFoods;
}

File: new-backend-app/Models/Meal.cs
namespace new_backend.Models;

public class Meal
{
    public virtual long Id { get; set; }
    public virtual long UserId { get; set; }
    public virtual string Title { get; set; } = null!;
    public virtual IList<MealFood> MealFoods { get; set; } = new List<MealFood>();

    public Meal()
    {
    }
}